File: js/Particle.js
/**
 * This class is for all the flying particles.
 */
class Particle {
    constructor() {
        this.w = Math.floor(Math.random() * 5) + 1; // width
        this.h = this.w; // height
        
        this.x = Math.floor(Math.random() * Way.w);
        this.y = Math.floor(Math.random() * Way.h);
        
        this.xV = Math.random() * 2 + 3; // velocity
        this.doubleX = this.x; // precise x coordinate
    }
    
    // Getters
    getXV() { return this.xV; }
    getX() { return this.x; }
    getY() { return this.y; }
    getWidth() { return this.w; }
    getHeight() { return this.h; }
    
    // Setters
    setX(xx) {
        this.x = xx;
        this.doubleX = xx;
    }
    
    setY(yy) { this.y = yy; }
    setXV(xv) { this.xV = xv; }
    
    /**
     * Draws the particle to the canvas context
     */
    draw(ctx) {
        ctx.fillStyle = '#444444'; // Dark gray
        ctx.fillRect(this.x, this.y, this.w, this.h);
    }
    
    /**
     * Updates the position
     */
    updatePosition() {
        this.doubleX -= this.xV;
        this.x = Math.floor(this.doubleX);
    }
}

File: Particle.java
/**
This class is for all the flying particles.
*/

import java.awt.*;

public class Particle
{
	private int w, h; //The width and height of Particle
	private int x, y; //The x and y coordinates Particle
	private double xV; //The current speed of Box in x-axis
	private double doubleX; //The decimal x coordinate

	public Particle()
	{
		w = (int)(Math.random()*5+1);
		h = w;

		x = (int)(Math.random()*Way.w);
		y = (int)(Math.random()*Way.h);

		xV = (Math.random()*2+3);

		doubleX = x;
	}

	/**
	Get/set methods
	*/
	//Returns xV
	public double getXV(){return xV;}

	//Returns x
	public int getX(){return x;}

	//Returns y
	public int getY(){return y;}

	//Returns width
	public int getWidth(){return w;}

	//Returns height
	public int getHeight(){return h;}

	//Sets x
	public void setX(int xx)
	{
		x = xx;
		doubleX = xx;
	}

	//Sets y
	public void setY(int yy){y = yy;}

	//Sets xV
	public void setXV(double xv){xV = xv;}

	/**
	Draws the box to a specified Graphics
	*/
	public void draw(Graphics g)
	{
		g.setColor(Color.DARK_GRAY);
		g.fillRect(x, y, w, h);
	}

	/**
	Updates the position
	*/
	public void updatePosition()
	{
		doubleX -= xV;
		x = (int)doubleX;
	}
}
